src/main/java/eu/dissco/core/translator/service/DwcaService.java

```diff
@@ -96,25 +99,35 @@ public void retrieveData() {
           .then().toFuture().get();
       archive = DwcFiles.fromCompressed(file, Path.of(dwcaProperties.getTempFolder()));
       var ids = postArchiveToDatabase(archive);
-      getSpecimenData(ids, archive);
+      getSpecimenData(ids, archive, processedRecords);
     } catch (IOException e) {
       log.error("Failed to open output stream for download file", e);
+      return new TranslatorJobResult(JobState.FAILED, processedRecords.get());
     } catch (ExecutionException e) {
       log.error("Failed during downloading file with exception", e.getCause());
+      return new TranslatorJobResult(JobState.FAILED, processedRecords.get());
     } catch (InterruptedException e) {
       log.error("Failed during downloading file due to interruption", e);
       Thread.currentThread().interrupt();
+      return new TranslatorJobResult(JobState.FAILED, processedRecords.get());
     } catch (DisscoRepositoryException e) {
       log.error("Failed during batch copy into temp tables with exception", e);
+      return new TranslatorJobResult(JobState.FAILED, processedRecords.get());
     } finally {
       if (archive != null) {
         log.info("Cleaning up database tables");
         removeTempTables(archive);
       }
     }
+    if (processedRecords.get() == 0) {
+      log.info("No records were successfully processed");
+      return new TranslatorJobResult(JobState.FAILED, processedRecords.get());
```

Comment:
A thought: if 99/100 jobs fail, it's marked as COMPLETED, but if 100/100 fail, its FAILED. Is that distinction helpful to us? Do we want to consider the possibility of a PARTIAL_SUCCESS job state? (Something to discuss for a new pr probably).

Reply:
Yes completely agree. Thought of this later, but I will create a task for this.https://naturalis.atlassian.net/browse/DD-1114